### PR TITLE
I53/university frontend

### PR DESCRIPTION
**Pull request by PJUllrich**
Added Positions, Applications and more.
This PR includes all functionality that Peter needs for his Masterthesis.

### CODE REVIEW

**Comment by pimotte** [comment]
ERROR in src/app/model/exchangePosition.ts(2,39): error TS2307: Cannot find module '../../../../university/src/app/model/schemaDefinition'.

Please copy that file over, as it can't run inside docker now xD

**Comment by pimotte** [comment]
If I apply using wrong credentials, the UI says I succeeded. 
The phrasing in the UI is a little bit weird. You create "Enrolment/Transcript" positions. On the "Create Position" dialog, I would add "Required proof" as hint to the <select>

Is creating proofs with empty fields supported (as in no filtering?). Trying to do this with an Enrolment yields: `java.lang.NullPointerException: Unknown proof: enrolmentproof`
With a Transcript: `
```backend-university_1  | java.lang.NullPointerException: null
backend-university_1  |         at nl.quintor.studybits.university.services.TranscriptProofService.handleProof(TranscriptProofService.java:38)
backend-university_1  |         at nl.quintor.studybits.university.services.TranscriptProofService.handleProof(TranscriptProofService.java:17)
backend-university_1  |         at nl.quintor.studybits.university.services.ProofHandler.handleProof(ProofHandler.java:118)
backend-university_1  |         at nl.quintor.studybits.university.services.ProofHandler$$FastClassBySpringCGLIB$$e0f6e325.invoke(<generated>)
```

File: student/src/app/components/connection/connection.component.html [hunk 1]
```diff
@@ -22,8 +22,8 @@
     <ng-container matColumnDef="confirmed">
       <mat-header-cell *matHeaderCellDef mat-sort-header>Identity confirmed</mat-header-cell>
       <mat-cell *matCellDef="let element">
-        <mat-icon class="color-green" *ngIf="element.confirmed">check_circle</mat-icon>
-        <mat-icon class="color-red" *ngIf="!element.confirmed">error</mat-icon>
+        <mat-icon class="icon_success" *ngIf="element.confirmed">check_circle_outline</mat-icon>
```
Comment (pimotte) [inline student/src/app/components/connection/connection.component.html]: Is there a way to actually be confirmed currently?

Comment (pimotte) [reply student/src/app/components/connection/connection.component.html]: Nvm, found the button

File: student/src/app/components/top-menu/top-menu.component.html [hunk 2]
```diff
@@ -14,9 +14,17 @@
     <mat-icon>settings_ethernet</mat-icon>
     Connections
   </button>
-  <button mat-button routerLink="/proof-requests">
+  <!--<button mat-button routerLink="/proof-requests">-->
```
Comment (pimotte) [inline student/src/app/components/top-menu/top-menu.component.html]: Please delete commented code

**Comment by pimotte** [comment]
Do you agree it would make sense to have a different theme for the university frontend. That way in demo's it's much clearer that it is in fact another application

**Comment by PJUllrich** [comment]
All comments (except for the empty fields in Position comment) were addressed in the latest commit.

**Comment by pimotte** [comment]
Changes look good:)

The empty fields comment is something that is relevant, I'd say. The Enrolment makes no sense without this functionality, and I imagine a demo with a Transcript where we ask all the field, but do not prefill the average. Otherwise you can only create a position for someone with a specific average.

Also, my formatting was messed up, and there was a comment in that word sea that you missed, I think:

The phrasing in the UI is a little bit weird. You create "Enrolment/Transcript" positions, but that's not really the name of the position. On the "Create Position" dialog, I would add "Required proof" as hint to the select/dropdown.

**Pull request merged**